src/libraries/Common/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Pkcs12.cs

```diff
@@ -355,48 +371,77 @@ attrType switch
             }
         }
 
+        private static void RejectDuplicateAttributes(AttributeAsn[] bagAttributes, HashSet<string> duplicateAttributeCheck)
+        {
+            // If there's only one attribute set there's no reason to instantiate the HashSet.
```

Comment:
This comment is out of date now re "instantiate the HashSet", but now that there is always a HashSet being passed in I wonder if the whole Length == 1 optimization could go away.